metaflow/plugins/resources_decorator.py

```diff
@@ -50,6 +50,16 @@ def myStep(self):
         KFP: Memory limit for this step. Default unit is MB.
             More memory units are supported, including "E", "P", "T", "G", "M", "K". (i.e. "4000M")
              Defaults None - relying on Kubernetes defaults.
+    local_storage: Union[int, str]
+        Not for AWS Batch.
+        KFP: Local ephemeral storage required.
+            This is local disk storage per step and lost after the step.
+            Default unit is MB.
+            More memory units are supported, including "E", "P", "T", "G", "M", "K". (i.e. "4000M")
+            Defaults None - relying on Kubernetes defaults.
+    local_storage_limit: Union[int, str]
```

Comment:
Here we've been modify the Metaflow `@resources` decorator in our fork, my assumption here is then that we're calling this new feature `local_storage` instead of `ephemeral_storage` as it's more generic and therefore more likely to work in the wider Metaflow ecosystem?

Reply:
True, it's also more descriptive.  Every step's resources are by default ephemeral (lost after the step completes).
I could have called it just storage, but it's really local disk storage.
Making space for coming soon `local_storage_volume`.  Thoughts?

Reply:
How would `local_storage_volume` be used? Is it a named volume that can be shared across steps (may be troublesome for retries), or is it just a fresh ephemeral_storage volume?

`storage` sounds better to be unless you want to distinguish and reserve it for the upcoming changes that you are considering. Customer doesn't need to know if it is local storage or ephemeral storage.

Reply:
Every resource used by a step is ephemeral after it has completed.  So adding ephemeral is redundant, and local means native to the container.

A volume is attached to the container.  So I think this is cleaner:
```python
@resources(local_storage=100, storage_volume="20G")
```
However, this PR is NOT about `storage_volume`, it's about `local_storage`, we can discuss the volume one separately.

Reply:
I see. Just clarifying is `local_storage` meant for low disk usage jobs to avoid excess volume attachment, while `storage_volume` enables jobs like spark or ZNN to store large chunks of data?

I think customers only uses one of the `local_storage` or `storage_volume` but not both. We can abstract this by using a single resource name `storage`, and make a resource choice inside workflow SDK. When we support volumes we can do 
```
if storage < 5000:  # 5 GB, subject to discussion
    <set local storage limit and requests>
else:
    <attach a volume>
```

Using `storage` over `local_storage` and `storage_volume` this way has the benefit of:
- customers not needing to distinguish two storage solutions, and not mistakenly using less efficient solutions like `local_storage = 500G` or `storage_volume = 2G`
- early customers do not need to change their resource specification code when we change the storage backend